graphql/schemabuilder/pagination.go

```diff
@@ -372,6 +380,62 @@ func (c *connectionContext) pagesFromEdges(edges []Edge, limit int) (pages []str
 	return pages
 }
 
+var matchGroups = regexp.MustCompile(`(?:([^\s"]+)|"([^"]*)"?)+`)
+
+// Split terms by spaces, except for spaces within quotes.
+// Like http://stackoverflow.com/questions/16261635/javascript-split-string-by-space-but-ignore-space-in-quotes-notice-not-to-spli
+// except:
+// - Treat the query ["san fran] as having the single term ["san fran"] instead of ["san", "fran"]
+// - Removes the delimiting quotes.
+func matchString(str, query string) bool {
+	matches := matchGroups.FindAllStringSubmatch(query, -1)
```

Comment:
This is returning  results in a different format when we use the quotes.  We have have three elements in the match instead of two.  The result seems to be that we start filtering on "" (which always has a match) so we won't get any filtering at all: https://play.golang.org/p/xlcQ73hpZth

Reply:
Should probably add a test for this case as well.

Reply:
Agreed - I think this is worth extracting into a separate internal package where it can be tested.

Reply:
Done